src/content/packages-and-plugins/swift-package-manager/for-app-developers.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Swift Package Manager for app developers
+description: How to use Swift Package Manager for native iOS or macOS dependencies
+---
+
+Flutter is migrating to Swift Package Manager to manage iOS and macOS native
+dependencies.
+This is an experimental feature that may change in the future.
+It is currently only available on the [`master` channel][].
+Flutter will continue to support CocoaPods until further notice.
+
+:::tip
+If you find a bug in Flutter's Swift Package Manager feature,
+please [open an issue][].
+:::
+
+[`master` channel]: /release/upgrade#switching-flutter-channels
+[open an issue]: {{site.github}}/flutter/flutter/issues/new?template=2_bug.yml
+
+{% include docs/swift-package-manager/how-to-enable-disable.md %}
+
+## Add Flutter to an existing app (add-to-app)
+
+Flutter's Swift Package Manager feature does not support add-to-app scenarios
+yet.
+
+For status updates, see [flutter#146957].
+
+[flutter#146957]: https://github.com/flutter/flutter/issues/146957
+
+## How to add Swift Package Manager integration to a project manually
+
+{% tabs %}
+{% tab "iOS project" %}
+
+{% include docs/swift-package-manager/migrate-ios-project-manually.md %}
+
+{% endtab %}
+{% tab "macOS project" %}
+
+{% include docs/swift-package-manager/migrate-macos-project-manually.md %}
+
+{% endtab %}
+{% endtabs %}
+
+## How to use a Swift Package Manager Flutter plugin that requires a higher OS version
+
+If a Swift Package Flutter Manager plugin requires a higher OS version than
+the project, you may get an error like this:
+
+```text
+Target Integrity (Xcode): The package product 'plugin_name_ios' requires minimum platform version 14.0 for the iOS platform, but this target supports 12.0
+```
+
+To use the plugin, you'll need to increase the Minimum Deployments of your
+app's target.
+
+:::warning
+This raises your app's minimum OS version requirement.
+:::
+
+{% render docs/captioned-image.liquid,
+image:"development/packages-and-plugins/swift-package-manager/minimum-deployments.png",
+caption:"The target's Minimum Deployments setting" %}
+
+## How to add Swift Package Manager integration to a custom target
+
+Follow the steps in
+`How to add Swift Package Manager integration to an iOS/macOS project manually`.
+
+In Step 1, list item 6 use your custom target instead of the Flutter target.
+
+In Step 2, list item 6 use your custom target instead of the Flutter target.
+
+## How to remove Swift Package Manager integration
```

Comment:
In theory no one needs to do these steps unless there's a horrible bug in our SwiftPM integration.